### ISSUE TITLE
bug(semantic): cannot create function declaration in if statement in non-strict code

### ISSUE THREAD

**Issue by DonIsaac** [oxc-project/oxc#4462]
Creating a function declaration within an if statement in non-strict code should be allowed. Right now, declarations in if statements with block statements are allowed, but without them they are incorrectly reported as an error.

```js
// no error, as intended
if (true) {
  function f() {}
}

// error, should be allowed
if (true) function g() {}
```
The error thrown is created  by this diagnostic function:

```rs
// crates/oxc_semantic/src/checker/javascript.rs, line 483
fn function_declaration_non_strict(span0: Span) -> OxcDiagnostic {
    OxcDiagnostic::error("Invalid function declaration")
.with_help("In non-strict mode code, functions can only be declared at top level, inside a block, or as the body of an if statement")
.with_label(span0)
}
```

**Comment by Boshen** [comment]
This is probably due to assuming the file is in `script` mode, where as we default everything to `module` https://github.com/oxc-project/oxc/blob/4c889977274c63847b11f655f0eb9e292a9a40b9/crates/oxc_span/src/source_type/mod.rs#L157

**Comment by overlookmotel** [comment]
Confirmed. Behavior of `oxc_semantic` is correct. If you use `SourceType::default()` (i.e. sloppy mode), semantic does not raise the error. In strict mode, it does.

**Comment by DonIsaac** [comment]
Both examples above are run in non-strict mode. The only difference is the presence of parenthesis
